color_refinement_helper.py

```diff
@@ -138,6 +141,82 @@ def get_twins(g: Graph):  # -> List[(Vertex, Vertex)], List[(Vertex, Vertex)]:
     return twins, false_twins
 
 
+def graph_to_modules(graph: Graph) -> ModularDecomposition:
+    vertices = graph.vertices
+    vertices_in_any_module = []
+    modular_decomposition = []
+
+    for vertex in vertices:
+        if vertex in vertices_in_any_module:
+            continue
+
+        module = [vertex]
+        vertices_in_any_module.append(vertex)
+
+        neighbours = vertex.neighbours
+
+        other_vertices = [vertex for vertex in vertices if vertex not in vertices_in_any_module]
+        for other_vertex in other_vertices:
+            other_neighbours = set(other_vertex.neighbours) - {vertex}
+            if other_neighbours == (set(neighbours) - {other_vertex}):
+                module.append(other_vertex)
+                vertices_in_any_module.append(other_vertex)
+
+        modular_decomposition.append(module)
+
+    return modular_decomposition
+
+
+def modules_to_graph(modules: ModularDecomposition) -> Graph:
+    """
+    Returns a new Graph object with Modules compressed to a Vertex
+    :param modules: list of modules
+    :return: new Graph
+    """
+    edges_list = []
+    for module in modules:
+        for vertex in module:
+            edges_list += get_edges_of_vertex(vertex)
+
+    for module in modules:
+        if len(module) > 1:
+            new_label = create_new_label(module)
+            edges_list = relabel_edges(module, edges_list, new_label)
+
+    edges = set()
+    for edge in edges_list:
+        if edge[0] != edge[1]:
```

Comment:
Checkt dit of edge.head != edge.tail? En kun je dat hebben?

Reply:
Zie 1 regel verder: `sorted(edges)`, met `edges` een set, garandeert dat.